File: Python/nes_exploration/event_logger.py
class EventLogger:
    def __init__(self):
        self.log = []

    def log_event(self, source: str, message: str):
        """Log an event from a given source."""
        entry = f"[{source}] {message}"
        self.log.append(entry)

    def get_recent(self, count: int = 5) -> list:
        """Return the most recent log entries."""
        return self.log[-count:]

File: nes_exploration_ts/eventLogger.ts
export class EventLogger {
  log: string[] = [];

  logEvent(source: string, message: string): void {
    /** Log an event from a given source. */
    const entry = `[${source}] ${message}`;
    this.log.push(entry);
  }

  getRecent(count: number = 5): string[] {
    /** Return the most recent log entries. */
    return this.log.slice(-count);
  }
}
